### PR TITLE
Coroutine

### PR DESCRIPTION
**Pull request by SongRe**

### CODE REVIEW

File: clang/lib/Parse/ParseDeclCXX.cpp [hunk 1]
```diff
@@ -1724,6 +1726,11 @@ void Parser::ParseClassSpecifier(tok::TokenKind TagTokKind,
     TagType = DeclSpec::TST_interface;
   else if (TagTokKind == tok::kw_class)
     TagType = DeclSpec::TST_class;
+  else if (TagTokKind == tok::kw__Coroutine) {
+    llvm::errs() << "_Coroutine recognized as class\n";
```
Comment (abdurj) [inline clang/lib/Parse/ParseDeclCXX.cpp]: nit: remove error msg

File: clang/lib/Sema/SemaCodeComplete.cpp [hunk 2]
```diff
@@ -1817,8 +1817,13 @@ static void AddTypeSpecifierResults(const LangOptions &LangOpts,
     Results.AddResult(
         Result("bool", CCP_Type + (LangOpts.ObjC ? CCD_bool_in_ObjC : 0)));
     Results.AddResult(Result("class", CCP_Type));
+    Results.AddResult(Result("_Coroutine", CCP_Type));
     Results.AddResult(Result("wchar_t", CCP_Type));
 
+    Builder.AddTypedTextChunk("_Coroutine");
+    Builder.AddChunk(CodeCompletionString::CK_HorizontalSpace);
+    Builder.AddInformativeChunk("_Coroutine test");
```
Comment (abdurj) [inline clang/lib/Sema/SemaCodeComplete.cpp]: what does this do

File: llvm-project.code-workspace
Comment (abdurj) [inline llvm-project.code-workspace:1]: uncommit this?

File: llvm/test/tools/opt-viewer/lit.local.cfg [hunk 3]
```diff
@@ -8,3 +8,5 @@ if "have_opt_viewer_modules" not in config.available_features:
 # can be resolved.
 if sys.platform == "win32":
     config.unsupported = True
+if sys.platform == 'darwin':
```
Comment (abdurj) [inline llvm/test/tools/opt-viewer/lit.local.cfg:11]: ?

File: clang/lib/Sema/SemaCodeComplete.cpp (hunk 2 above)
Comment (SongRe) [reply clang/lib/Sema/SemaCodeComplete.cpp]: This gives us the information when you hover over the keyword

File: llvm-project.code-workspace
Comment (SongRe) [reply llvm-project.code-workspace:1]: ok

File: llvm/test/tools/opt-viewer/lit.local.cfg (hunk 3 above)
Comment (SongRe) [reply llvm/test/tools/opt-viewer/lit.local.cfg:11]: disables a known failing test on MacOS

**Review by abdurj — APPROVED** [review]
lgtm

File: clang-tools-extra/clangd/ClangdLSPServer.cpp
Comment (abdurj) [inline clang-tools-extra/clangd/ClangdLSPServer.cpp:1]: revert ?

**Review by abdurj — APPROVED** [review]
lgtm

**Pull request merged**